esmf-semantic-aspect-meta-model/src/main/resources/samm/meta-model/2.2.0/aspect-meta-model-shapes.ttl

```diff
@@ -499,6 +499,19 @@ samm:ConstraintShape
       sh:maxCount 0 ;
       sh:name "dataType" ;
       sh:description "A Constraint may not change the data type of the Characteristic being constrained." ;
+   ] ;
+   sh:sparql [
+      a sh:SPARQLConstraint ;
+      sh:message "Constraint '{$this}' must have a baseCharacteristic or a dataType." ;
```

Comment:
Constraints never have a baseCharacteristic (only Traits do), so this message is confusing. How about:
```suggestion
      sh:message "Constraint '{$this}' has invalid type samm:Constraint, only subtypes of samm:Constraint may be used." ;
```